cmd/vela-makisu/build.go

```diff
@@ -4,15 +4,561 @@
+		if err != nil {
+			return err
+		}
+	}
+
+	// check if any redis options were passed
+	if len(b.RedisCacheRaw) > 0 {
+		// cast raw http options into bytes
+		redisOpts := []byte(b.RedisCacheRaw)
+
+		// serialize raw http options into expected RedisCache type
+		err := json.Unmarshal(redisOpts, &b.RedisCache)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
 }
 
 // Validate verifies the Build is properly configured.
 func (b *Build) Validate() error {
 	logrus.Trace("validating build plugin configuration")
 
+	// verify tag are provided
+	if len(b.Context) == 0 {
+		return fmt.Errorf("no build context provided")
+	}
+
+	// verify tag are provided
+	if len(b.Tag) == 0 {
+		return fmt.Errorf("no build tag provided")
+	}
+
 	return nil
 }
+
+// Flags formats and outputs the flags for
+// configuring a Docker.
+func (d *Docker) Flags() []string {
+	// variable to store flags for command
+	var flags []string
+
+	// check if Host is provided
+	if len(d.Host) > 0 {
+		// add flag for Host from provided build command
+		flags = append(flags, fmt.Sprintf("--docker-host=%s", d.Host))
+	}
+
+	// check if Scheme is provided
+	if len(d.Scheme) > 0 {
+		// add flag for Scheme from provided build command
+		flags = append(flags, fmt.Sprintf("--docker-scheme=%s", d.Scheme))
+	}
+
+	// check if Version is provided
+	if len(d.Version) > 0 {
+		// add flag for Version from provided build command
+		flags = append(flags, fmt.Sprintf("--docker-version=%s", d.Version))
+	}
+
+	return flags
+}
+
+// Flags formats and outputs the flags for
+// configuring a http cache.
+func (h *HTTPCache) Flags() []string {
+	// variable to store flags for command
+	var flags []string
+
+	// check if Addr is provided
+	if len(h.Addr) > 0 {
+		// add flag for Addr from provided build command
+		flags = append(flags, fmt.Sprintf("--http-cache-addr=%s", h.Addr))
+	}
+
+	// check if Headers is provided
+	if len(h.Headers) > 0 {
+		var args string
+		for _, arg := range h.Headers {
+			args += fmt.Sprintf(" %s", arg)
+		}
+		// add flag for BuildArgs from provided build command
+		flags = append(flags, fmt.Sprintf("--http-cache-header=%s", strings.TrimPrefix(args, " ")))
+	}
+
+	return flags
+}
+
+// Flags formats and outputs the flags for
+// configuring a redis cache.
+func (r *RedisCache) Flags() ([]string, error) {
+	// variable to store flags for command
+	var flags []string
+
+	// check if Addr is provided
+	if len(r.Addr) > 0 {
+		// add flag for Addr from provided build command
+		flags = append(flags, fmt.Sprintf("--redis-cache-addr=%s", r.Addr))
+	}
+
+	// check if Password is provided
+	if len(r.Password) > 0 {
+		// add flag for Password from provided build command
+		flags = append(flags, fmt.Sprintf("--redis-cache-password=%s", r.Password))
+	}
+
+	// check if TTL is provided
+	if len(r.TTL) > 0 {
+		duration, err := time.ParseDuration(r.TTL)
+		if err != nil {
+			return nil, err
+		}
+
+		// check if TTL is valid duration
+		if !isDurationZero(duration) {
```

Comment:
I don't think we need to call a function here to check if the duration is zero 👍 

`time.Duration` has a type of `int64` so we should be able to compare against `0` itself

```suggestion
		if duration > 0 {
```

Reply:
Yeah, I was wondering how to do 0 value with duration. I tried doing some searches and didn't find a ton I liked so I wrote my own. I can look at simplifying this with your suggestions.